partner_invoicing_mode_fourteen_days/tests/test_invoicing_fourteen_days.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from freezegun import freeze_time
+
+from odoo.fields import Datetime
+from odoo.tests.common import TransactionCase
+
+from odoo.addons.partner_invoicing_mode.tests.common import CommonPartnerInvoicingMode
+from odoo.addons.queue_job.tests.common import trap_jobs
+
+
+class TestInvoiceModeTenDays(CommonPartnerInvoicingMode, TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner.invoicing_mode = "fourteen_days"
+        cls.partner2.invoicing_mode = "fourteen_days"
+
+    def test_invoice_mode(self):
+        """
+        Try to invoice the concerned sale orders
+        """
+        self._confirm_and_deliver(self.so1)
+        self.assertFalse(self.so1.invoice_ids)
+        with trap_jobs() as traps:
+            self.SaleOrder.generate_fourteen_days_invoices()
+            traps.enqueued_jobs[0].perform()
+        self.assertTrue(self.so1.invoice_ids)
+
+    def test_invoice_mode_ten_days_only(self):
+        """
+        Try to invoice the concerned sale orders
+        """
+        self.partner2.invoicing_mode = "standard"
+        so3 = self.so2.copy(
+            {
+                "partner_id": self.partner2.id,
+                "partner_invoice_id": self.partner2.id,
+                "partner_shipping_id": self.partner2.id,
+            }
+        )
+        self._confirm_and_deliver(self.so1)
+        self._confirm_and_deliver(so3)
+        self.assertFalse(self.so1.invoice_ids)
+        self.assertFalse(so3.invoice_ids)
+        with trap_jobs() as traps:
+            self.SaleOrder.generate_fourteen_days_invoices()
+            traps.enqueued_jobs[0].perform()
+        self.assertTrue(self.so1.invoice_ids)
+        self.assertFalse(so3.invoice_ids)
+
+    def test_cron(self):
+        with freeze_time("2023-06-01"):
+            self._confirm_and_deliver(self.so1)
+        with freeze_time("2023-06-10"):
+            with trap_jobs() as trap:
+                self.env["sale.order"].cron_generate_fourteen_days_invoices()
+                trap.assert_jobs_count(1)
+                trap.assert_enqueued_job(
+                    self.env["sale.order"]._generate_invoices_by_partner,
+                    args=(self.so1.ids,),
+                    kwargs={},
+                )
+                with trap_jobs() as trap_invoice:
+                    for job in trap.enqueued_jobs:
+                        if job.args == (self.so1.ids,):
+                            job.perform()
+                    trap_invoice.assert_jobs_count(1)
+                    trap_invoice.assert_enqueued_job(
+                        self.so1.invoice_ids._validate_invoice,
+                        args=(),
+                        kwargs={},
+                    )
+                    trap_invoice.enqueued_jobs[0].perform()
```

Comment:
If more jobs have been enqueued in the meanwhile there is a risk of performing a wrong one by using the index 0. Maybe consider calling `trap_invoice.perfom_enqueued_jobs()`